File: backend/internal/dto/documents.go
package dto

type PrintHeader struct {
	ClinicName        string `json:"clinic_name"`
	DocumentTitle     string `json:"document_title"`
	ConsultationID    string `json:"consultation_id"`
	PatientName       string `json:"patient_name"`
	PatientIdentifier string `json:"patient_identifier"`
	DoctorName        string `json:"doctor_name"`
	Date              string `json:"date"`
}

type PrescriptionDocument struct {
	Header      PrintHeader     `json:"header"`
	Medications []MedicationDTO `json:"medications"`
}

type LabRequestDocument struct {
	Header   PrintHeader `json:"header"`
	LabTests []LabTestDTO `json:"lab_tests"`
}

type NotesDocument struct {
	Header        PrintHeader      `json:"header"`
	ClinicalNotes ClinicalNotesDTO `json:"clinical_notes"`
}

type BillDocument struct {
	Header  PrintHeader       `json:"header"`
	Billing BillingSummaryDTO `json:"billing"`
}

File: frontend/src/types/documents.ts
import { BillingSummary, ClinicalNotes, LabTest, Medication, PrintHeader } from "./api";

export type PrescriptionDocument = {
  header: PrintHeader;
  medications: Medication[];
};

export type LabRequestDocument = {
  header: PrintHeader;
  lab_tests: LabTest[];
};

export type NotesDocument = {
  header: PrintHeader;
  clinical_notes: ClinicalNotes;
};

export type BillDocument = {
  header: PrintHeader;
  billing: BillingSummary;
};
